### DISCUSSION TITLE
Group 14 Discussion

### DISCUSSION THREAD

**Discussion by asegal20902** [Digital-Preservation-Coalition/PythonStudyGroups#23] [General]
Hi all Group 14 members! I have been very busy with a paper proposal (which was just accepted :) ) but I would like to start discussion with our group on Github. I plan to work intensively on the Python tutorial next weekend (Presidents' Day in the US, so I have that Monday off). I'd like to start things off with a slide I produced during the pandemic on how I used Python at the Library of Congress during a learning group to use the Python Library "Beautiful Soup" to retrieve Library data about the papers of President William Howard Taft, which I helped get online, and about a book of poetry by great weird fiction author Clark Ashton Smith, because I am interested in science fiction and fantasy ;) . I would also like to share the link from Digital Transitions, a camera company which produces imaging cameras and software that the Library of Congress uses, to a webinar about how to use Python. https://www.youtube.com/watch?v=mozYgcGzjqs
[Showcase-Show-Tell.pptx](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/files/14231178/Showcase-Show-Tell.pptx) 
Perhaps we can discuss more of what we are doing tomorrow or next weekend. 
Best Ariel Segal

**Comment by asegal20902** [comment]
By the end of Presidents' Day weekend, I am planning to 
- Read through the primer https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/wiki
- Go through the Digital Transitions Webinar on Python using Visual Studio Code
- Read through the "Working with APIs" use case https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20-%20Working%20with%20apis.md and try to make it work using Visual Studio Code. 
Does that sound reasonable for the rest of us to do? Any other suggestions? 
Thanks Ariel

**Comment by digitensions** [comment]
Excellent stuff! If you have any code issues you might want support with don't forget to give a mod a shout. Lots of luck!

**Comment by asegal20902** [comment]
Hi I am working through the Digital Transitions Python Course from 2020 on YouTube. I may need to work next Sunday to complete everything I am trying to do. Here's a link to the examples and images they provided, available on my Google Drive. https://drive.google.com/drive/folders/149MkvZd-JgM6O2aqK1WjgkokhooHlXZH?usp=sharing

**Comment by asegal20902** [comment]
@digitensions I am trying an import image module from the Python training video I did in 2020. It works successfully for most images on my desktop, but the TIFF images the instructors provided 
![PythonImportImageScreenshotIssue](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/assets/37121494/7de107d3-3576-40a5-ba1b-cd42147f6f76)
are returning an error as they are converted into temporary PNGs (see Screencap). I have tried the program with other TIFF images, jp2 images, jpg images on my computer and the program itself works. The TIFFs themselves open with no problem in the Mac Preview application. Any ideas as to the image error source?

**Comment by asegal20902** [comment]
A second module does successfully retrieve the size of the image even though it can't be displayed in the first module.

**Reply by digitensions** [reply]
Hey, are they 16-bit images? As you've found I think some modules PIL image io.read() won't work with the 16-bit tiff files.

There's a similar error discussed here with solution that may help:

https://stackoverflow.com/questions/43143562/io-imread-reading-some-tif-files-but-not-others

**Reply by digitensions** [reply]
I have a script which uses GraphicsMagick to handle image conversions, from TIF to JPEG with rescaling (thumbnail creation during proxy MP4 creation code). Unsure what you need to do with your images in the code, but it's another route you might want to take.
http://www.graphicsmagick.org/
This software works from the CLI, so you can use Python subprocess to launch it to do complex tasks. There are also python bindings:
https://pypi.org/project/pgmagick/

**Comment by asegal20902** [comment]
Hi. I went through the Digital Transitions Python video, which had exercises resizing and exporting TIFF images as jpgs, and renaming jpgs as recto/verso. It refreshed my memory pretty well. I will do the Working with APIs exercise next week after turning in a conference paper.
